common/cache/domainCache.go

```diff
@@ -86,50 +87,54 @@ func newDomainCacheEntry() *domainCacheEntry {
 
 // GetDomain retrieves the information from the cache if it exists, otherwise retrieves the information from metadata
 // store and writes it to the cache with an expiry before returning back
-func (c *domainCache) GetDomain(name string) (*persistence.DomainInfo, *persistence.DomainConfig, error) {
+func (c *domainCache) GetDomain(name string) (*domainCacheEntry, error) {
 	return c.getDomain(name, "", name, c.cacheByName)
 }
 
 // GetDomainByID retrieves the information from the cache if it exists, otherwise retrieves the information from metadata
 // store and writes it to the cache with an expiry before returning back
-func (c *domainCache) GetDomainByID(id string) (*persistence.DomainInfo, *persistence.DomainConfig, error) {
+func (c *domainCache) GetDomainByID(id string) (*domainCacheEntry, error) {
 	return c.getDomain(id, id, "", c.cacheByID)
 }
 
 // GetDomainID retrieves domainID by using GetDomain
 func (c *domainCache) GetDomainID(name string) (string, error) {
-	info, _, err := c.GetDomain(name)
+	entry, err := c.GetDomain(name)
 	if err != nil {
 		return "", err
 	}
-	return info.ID, nil
+	return entry.Info.ID, nil
 }
 
 // GetDomain retrieves the information from the cache if it exists, otherwise retrieves the information from metadata
 // store and writes it to the cache with an expiry before returning back
-func (c *domainCache) getDomain(key, id, name string, cache Cache) (*persistence.DomainInfo, *persistence.DomainConfig, error) {
+func (c *domainCache) getDomain(key, id, name string, cache Cache) (*domainCacheEntry, error) {
 	now := c.timeSource.Now().UnixNano()
-	refreshCache := false
-	var info *persistence.DomainInfo
-	var config *persistence.DomainConfig
+	var result *domainCacheEntry
+
+	copyEntry := func(input *domainCacheEntry) *domainCacheEntry {
```

Comment:
Would it be better if put `copyEntry` and `isCacheExpired` outside?  (to save some cycles on definition)